File: app/src/main/java/com/example/vincentwang/rainbowtranslate/store/SearchStorePresenter.java
package com.example.vincentwang.rainbowtranslate.store;

import com.example.vincentwang.rainbowtranslate.data.WordMain;
import com.example.vincentwang.rainbowtranslate.framework.BasePresenter;
import com.example.vincentwang.rainbowtranslate.translate.TranslateContract;

import java.util.ArrayList;
import java.util.Calendar;

import io.reactivex.subscribers.ResourceSubscriber;

/**
 * Created by vincentwang on 2017/8/27.
 */

public class SearchStorePresenter extends BasePresenter implements SearchStoreContract.Presenter {

    SearchStoreContract.View view;

    TranslateContract.Model model;

    public SearchStorePresenter(SearchStoreContract.View view, TranslateContract.Model model) {
        this.view = view;
        this.model = model;
    }


    @Override
    public void searchbuttonclick(int spinnerposition,Calendar startDay,Calendar endDay) {

        model.getWordMain(spinnerposition,startDay,endDay)
                .compose(this.<ArrayList<WordMain>>applySchedulers())
                .subscribeWith(new ResourceSubscriber<ArrayList<WordMain>>() {
                    @Override
                    public void onNext(ArrayList<WordMain> wordMains) {
                        view.showSearchList(wordMains);
                    }

                    @Override
                    public void onError(Throwable t) {

                    }

                    @Override
                    public void onComplete() {

                    }
                });
    }
}

File: app/src/main/java/com/example/vincentwang/rainbowtranslate/store/SearchStorePresenter.kt
package com.example.vincentwang.rainbowtranslate.store

import com.example.vincentwang.rainbowtranslate.data.WordMain
import com.example.vincentwang.rainbowtranslate.framework.BasePresenter
import com.example.vincentwang.rainbowtranslate.translate.TranslateContract
import io.reactivex.subscribers.ResourceSubscriber
import java.util.*

/**
 * Created by vincentwang on 2017/8/27.
 */

class SearchStorePresenter(var view: SearchStoreContract.View, var model: TranslateContract.Model) : BasePresenter(), SearchStoreContract.Presenter {
    override fun searchPeriod(startDay: Calendar?, endDay: Calendar?) {
        if (startDay != null && endDay != null && startDay.before(endDay)) {
            model.getWordMainPeriod(startDay, endDay)
                    .compose(applySchedulers())
                    .subscribeWith(object : ResourceSubscriber<ArrayList<WordMain>>() {
                        override fun onNext(wordMains: ArrayList<WordMain>) {
                            view.showSearchList(wordMains)
                        }

                        override fun onError(t: Throwable) {
                            view.showDialog("查詢","查無資料")
                        }

                        override fun onComplete() {

                        }
                    })
        }else
        {
            view.showDialog("查無資料","日期錯誤")
        }
    }

    override fun searchToday() {
        model.wordMainToday
                .compose(applySchedulers())
                .subscribeWith(object : ResourceSubscriber<ArrayList<WordMain>>() {
                    override fun onNext(wordMains: ArrayList<WordMain>) {
                        view.showSearchList(wordMains)
                    }

                    override fun onError(t: Throwable) {
                        view.showDialog("查詢","查無資料")
                    }

                    override fun onComplete() {

                    }
                })
    }

    override fun searchAll() {
        model.wordMainAll
                .compose(applySchedulers())
                .subscribeWith(object : ResourceSubscriber<ArrayList<WordMain>>() {
                    override fun onNext(wordMains: ArrayList<WordMain>) {
                        view.showSearchList(wordMains)
                    }

                    override fun onError(t: Throwable) {
                        view.showDialog("查詢","查無資料")
                    }

                    override fun onComplete() {

                    }
                })
    }
}
